test/chi0.jl

```diff
@@ -90,12 +95,16 @@ function test_chi0(testcase; symmetry=false, use_symmetry=false, temperature=0,
     end
 end
 
-for testcase in (silicon, magnesium)
-    temp = isnothing(testcase.temperature) ? (0, 0.03) : (testcase.temperature)
-    for temperature in temp, spin_polarization in (:none, :collinear)
+for (case, temperatures) in [(silicon, (0, 0.03)), (magnesium, (0.01, ))]
+    for temperature in temperatures, spin_polarization in (:none, :collinear)
         for use_symmetry in (false, true), symmetry in (false, true)
-            test_chi0(testcase; symmetry, use_symmetry,
-                      temperature, spin_polarization)
+            test_chi0(case; symmetry, use_symmetry, temperature, spin_polarization)
+            # additional test for compute_χ0 in some specific cases
+            if iszero(temperature) && !symmetry
+                test_chi0(case; symmetry, use_symmetry, temperature, spin_polarization,
+                          eigensolver=diag_full, Ecut=3, fft_size=[10, 1, 10],
+                          compute_full_χ0=true)
+            end
```

Comment:
Just make another for loop block for this, I think that is less confusing.

Reply:
what do you mean by "another loop block" ? on the flag being true or false ?

Reply:
what i mean is instead of "filtering out" the cases with `iszero(temperature) && !symmetry`, just have another loop only for the silicon test case where you run the modified `test_chi0` with the `compute_full_χ0=true`.